### PR TITLE
Sync ui integration

### PR DESCRIPTION
**Pull request by wtrocki**
## Motivation

See: https://github.com/feedhenry-raincatcher/raincatcher-angularjs/pull/11/

### CODE REVIEW

File: demo/mobile/src/app/modules/sync/syncPoolService.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+};
+
+/**
+ * Service to manage init all of the sync data sets.
+ *
+ * @param $q
+ * @returns {{}}
+ * @constructor
+ */
+function SyncPoolService($q) {
```
Comment (paolobueno) [inline demo/mobile/src/app/modules/sync/syncPoolService.js:76]: [OPTIONAL] angular's $q isn't being used here, we could refactor into bluebird or use it instead

File: demo/mobile/src/app/services/userService.js [hunk 2]
```diff
@@ -18,8 +18,13 @@ UserService.prototype.readUser = function readUser(userId) {
...
 };
 
+
+UserService.prototype.getProfile = function(userId) {
+  return this.readUser(userId);
+};
+
 UserService.prototype.listUsers = function listUsers() {
-  return Promise.resolve([]);
+  return Promise.all(this.readUser(userId));
```
Comment (paolobueno) [inline demo/mobile/src/app/services/userService.js:27]: Ah, using `Bluebird.all` to wrap the results in an array? That's actually nice  :+1:

Comment (wtrocki) [reply demo/mobile/src/app/services/userService.js:27]: Just temporary solution.

**Review by paolobueno — DISMISSED** [review]
reviewed, tested locally.

File: demo/mobile/src/app/modules/sync/syncDataRepository.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
+var Promise = require("bluebird");
+
+// Service that is being injected to modules to provide all operations around sync
+function SyncApiDataService(datasetId, syncService) {
+  this.syncManagerPromise = syncService.then(function (managers) {
+    return Promise.promisifyAll(managers[datasetId]);
```
Comment (paolobueno) [inline demo/mobile/src/app/modules/sync/syncDataRepository.js:6]: This one can't be ported to TS without some refactoring, but that's okay.

Alternative is promisifying methods individually, but more verbose: https://github.com/feedhenry-raincatcher/raincatcher-angularjs/pull/11/files#diff-8e1a8ec6edc1f75011072cb1f15d9cb0R7

File: demo/mobile/src/app/modules/sync/syncServices.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  return userService.getProfile()
+    .then(syncPool.syncManagerMap);
+}]);
+
+// Generating commom data repositories
+var SyncApiDataService = require("./syncDataRepository");
+
+var datasets = config.datasetIds;
+angular.module('wfm.common.apiservices').service("workorderService", ['syncService', function (syncService) {
+  return new SyncApiDataService(datasets.workorders, syncService);
```
Comment (paolobueno) [inline demo/mobile/src/app/modules/sync/syncServices.js:12]: :+1: very nice abstraction

**Review by paolobueno — APPROVED** [review]

**Pull request merged**